### PR TITLE
feat(generator): golden files for Auth decorators

### PR DESCRIPTION
**Pull request by coryan**
Add the golden files for Auth decorators and their unit tests. Note that
the generator does not automatically generate these files yet, so they
are not tested either.

Part of the work for #6721

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6749)

### CODE REVIEW

File: generator/integration_tests/golden/internal/golden_thing_admin_auth_decorator.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+class GoldenThingAdminAuth : public GoldenThingAdminStub {
+ public:
+  ~GoldenThingAdminAuth() override = default;
+  GoldenThingAdminAuth(
+      std::shared_ptr<google::cloud::internal::GrpcAuthenticationStrategy> auth,
+      std::shared_ptr<GoldenThingAdminStub> child);
+
+
```
Comment (devbww) [inline generator/integration_tests/golden/internal/golden_thing_admin_auth_decorator.h:44]: Nit: Did we intend to generate multiple newlines?

File: generator/integration_tests/golden/internal/golden_kitchen_sink_auth_decorator.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    google::test::admin::database::v1::ListServiceAccountKeysRequest const& request) {
+  auto status = auth_->ConfigureContext(context);
+  if (!status.ok()) return status;
+  return child_->ListServiceAccountKeys(context, request);
+}
+}  // namespace GOOGLE_CLOUD_CPP_GENERATED_NS
+}  // namespace golden_internal
+}  // namespace cloud
+}  // namespace google
+
```
Comment (devbww) [inline generator/integration_tests/golden/internal/golden_kitchen_sink_auth_decorator.cc:89]: Nit: Trailing blank line.

File: generator/integration_tests/golden/internal/golden_thing_admin_auth_decorator.h (hunk 1 above)
Comment (coryan) [reply generator/integration_tests/golden/internal/golden_thing_admin_auth_decorator.h:44]: I am going to punt on these to #5821, in general the code generated in `google/cloud/*` gets the `clang-format` treatment. The code in `generator/integration_tests/*` does not (yet).

File: generator/integration_tests/golden/internal/golden_kitchen_sink_auth_decorator.cc (hunk 2 above)
Comment (coryan) [reply generator/integration_tests/golden/internal/golden_kitchen_sink_auth_decorator.cc:89]: Ditto.

**Comment by coryan** [comment]
Ping

**Review by devjgm — APPROVED** [review]

**Pull request merged**